cmd/snap-bootstrap/cmd_initramfs_mounts.go

```diff
@@ -1546,20 +1562,22 @@ func generateMountsModeRun(mst *initramfsMountsState) error {
 
 	// TODO: do we actually need fsck if we are mounting a mapper device?
 	// probably not?
-	// fsck and mount with nosuid to prevent snaps from being able to bypass
-	// the sandbox by creating suid root files there and trying to escape the
-	// sandbox
 	dataMountOpts := &systemdMountOptions{
 		NeedsFsck: true,
-		NoSuid:    true,
-		Private:   true,
+	}
+	if !isClassic {
+		// fsck and mount with nosuid to prevent snaps from being able to bypass
+		// the sandbox by creating suid root files there and trying to escape the
+		// sandbox
+		dataMountOpts.NoSuid = true
+		dataMountOpts.Private = true
```

Comment:
this doesn't comment why we also don't want Private true

Reply:
I've added a comment. I don't use private here because I think that in general the behavior the user sees should be as close as possible to a regular classic system.